### DISCUSSION TITLE
Let's Generate a List of Aliases for Existing Math AutoCorrect Codes

### DISCUSSION THREAD

**Discussion by brichwin** [daisy/math-a11y#25] [General]
During Murray's demo of his auto-complete feature for entering math AutoCorrect codes, it was mentioned that some Math AutoCorrect codes do not begin with the obvious letters and the auto-complete would not help users to enter or to discover them. For example, the Math AutoCorrect code for the union symbol (∪) is \cup.

To help address this, Murray suggested it would be possible to add more Math AutoCorrect codes and that we should create a list. This discussion post can serve as a place where we can built a list. Once we have a firm list, we can create an issue with the list for Microsoft to consider.

**Note**: According to the MS Word documentation, when Math AutoCorrect codes conflict with the non-math AutoCorrect codes the non-math AutoCorrect codes take precedence. It seems unlikely, but we should check that the codes we suggest are not in the non-math AutoCorrect codes.

**Comment by brichwin** [comment]
From Jennifer Marsala
# Recommended Rich Edit Additions for Math Symbols in MS Word

At the 9/6/24 Math Writing Solutions meeting, we talked about entries
being added to the Rich Edit table of symbol commands in Word.

Below is a list of suggested additions.

<table>
<caption><p><strong>Suggested Rich Edit Additions for Math Symbols in MS
Word</strong></p></caption>
<colgroup>
<col style="width: 13%" />
<col style="width: 16%" />
<col style="width: 19%" />
<col style="width: 20%" />
<col style="width: 30%" />
</colgroup>
<thead>
<tr>
<th><strong>Symbol Name</strong></th>
<th><strong>Symbol</strong></th>
<th><strong>Unicode command</strong></th>
<th><strong>Recommended Addition(s)</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr>
<td>Composition</td>
<td><span class="math display">∘</span></td>
<td>\circ</td>
<td>\comp</td>
<td></td>
</tr>
<tr>
<td>Element</td>
<td><span class="math display">∈</span></td>
<td>\in</td>
<td>\element</td>
<td></td>
</tr>
<tr>
<td>Fraction</td>
<td><span class="math display">$$\frac{}{}$$</span></td>
<td>/</td>
<td>\frac</td>
<td>Perhaps this is not necessary since it’s so easy to type the
fraction symbol, but many people are used to typing \frac in LaTeX and
would likely try it in Unicode mode.</td>
</tr>
<tr>
<td>Infinity</td>
<td><span class="math display">∞</span></td>
<td>\infty</td>
<td>\infinity</td>
<td></td>
</tr>
<tr>
<td>Intersection</td>
<td><span class="math display">∩</span></td>
<td>\cap</td>
<td>\intersect<br />
\intersection</td>
<td></td>
</tr>
<tr>
<td>Line symbol</td>
<td><span
class="math display">$$\overset{\overleftrightarrow{}}{AB}$$</span></td>
<td>Line symbol was chosen from the Accent menu of the toolbar. (Unicode
symbol not known.) AB was added for clarity.</td>
<td>\line</td>
<td></td>
</tr>
<tr>
<td>Ray</td>
<td><span class="math display">$$\overrightarrow{AB}$$</span></td>
<td>\vec<br />
<br />
At left, (AB)\vec; spacebar must be pressed twice for proper
rendering.</td>
<td>\ray</td>
<td>Prior to PreCalculus, many students will know what a ray symbol is,
but not know anything about vectors (and would not think of using the
\vec command).</td>
</tr>
<tr>
<td>Roots</td>
<td><p><span class="math display">$$\sqrt{}$$</span></p>
<p><span class="math display">$$\sqrt[3]{}$$</span></p>
<p><span class="math display">$$\sqrt[4]{}$$</span></p>
<p><span class="math display">$$\sqrt[5]{r}$$</span></p>
<p><span class="math display">$$\sqrt[n]{r}$$</span></p></td>
<td><p>\sqrt<br />
\cbrt<br />
\qdrt<br />
\sqrt(5&amp;r), where r represents the radicand.</p>
<p><span class="math inline">$\sqrt[n]{r}$</span> is typed as
\sqrt(n&amp;r)</p></td>
<td>\2root<br />
\3root<br />
\4root<br />
\5root<br />
etc.</td>
<td>How many entries would be desirable? Maybe 2<sup>nd</sup> through
9<sup>th</sup> roots? Or if it’s programmable to include any root, like
\24root, that would be even more flexible.</td>
</tr>
<tr>
<td>Segment</td>
<td><span class="math display">$${\overline{AB}
}\overline{AB}$$</span></td>
<td>\overbar<br />
\overline<br />
<br />
Shown: \overbar(AB) and \overline(AB)</td>
<td>\seg<br />
\segment</td>
<td></td>
</tr>
<tr>
<td>Set of integers</td>
<td><span class="math display">ℤ</span></td>
<td>\doubleZ</td>
<td>\integer<br />
\integers</td>
<td></td>
</tr>
<tr>
<td>Set of natural numbers</td>
<td><span class="math display">ℕ</span></td>
<td>\doubleN</td>
<td>\natural<br />
\naturals</td>
<td></td>
</tr>
<tr>
<td>Set of rational numbers</td>
<td><span class="math display">ℚ</span></td>
<td>\doubleQ</td>
<td>\rational<br />
\rationals</td>
<td></td>
</tr>
<tr>
<td>Set of real numbers</td>
<td><span class="math display">ℝ</span></td>
<td>\doubleR</td>
<td>\real<br />
\reals</td>
<td></td>
</tr>
<tr>
<td>Triangle</td>
<td><span class="math display">△</span></td>
<td>This triangle symbol exists in the Operators category of the toolbar
(under Advanced Binary Operators) or can be typed in LaTeX mode using
the \triangle command. (Unicode symbol not known.)</td>
<td>\triangle</td>
<td></td>
</tr>
<tr>
<td>Union</td>
<td><span class="math display">∪</span></td>
<td>\cup</td>
<td>\union</td>
<td></td>
</tr>
</tbody>
</table>

**Reply by jenmarsala** [reply]
Here are some more suggestions (edited so that all math symbols display correctly):

**Additional Suggested Aliases for Math AutoCorrect Codes**

Symbol    Name | Symbol | Current    Autocorrect code | Recommended    autocorrect code | Notes
-- | -- | -- | -- | --
Parallelogram | &#x25b1;  | \underline | \par<br>   \parallelogram | Note:   \parallel creates the parallel symbol, .
Arc   symbol | $\overparen{AB}$ | \overparen | \arc |  
Circle with   dot inside | $\odot$  | \odot | \circledot |  
Logical   and (conjunction) | $\wedge$  | \wedge | \and |  
Logical   or (disjunction) | $\vee$ | \vee | \or |  
Logical   not | $\neg$ | \neg | \not |  
Exclusive   disjunction / exclusive or | $\oplus$<br> $\bigoplus$ | \oplus   \bigoplus | \xor | We need   to decide which of the two (very similar) symbols to use. \oplus reads as “circle   plus” and \bigoplus reads as “circle plus operator.”
Degree | $\degree$ | \degree | \deg | This   isn’t necessary but provides a shorter abbreviation.
Double   integral | $\iint$ | \iint | \doubleint |  
Triple   integral | $\iiint$ | \iiint | \tripleint |  
Double prime | $\prime \prime$ | \pprime | \dprime<br>   \doubleprime |  
Triple prime | $\prime \prime \prime$ | \ppprime | \tprime<br>   \tripleprime |  
Quadruple prime | $\prime \prime \prime \prime$  | \pppprime | \qprime<br>   \quadprime |

**Reply by jenmarsala** [reply]
Based on a discussion from <a href="https://github.com/daisy/math-a11y/discussions/26">Discussion Item 26 - Let's Generate a List of Missing Math AutoCorrect Codes for Commonly Used Symbols</a>... If all are in agreement, let's remove the additional suggested codes  in the table above (which I originally gave!) for the sets of integers, natural numbers, rational numbers, and real numbers. Let's just stick with the existing codes of \doubleZ, \doubleN, etc. (and NOT suggest codes like \integers or \naturals) because as Brian suggested, there is instructional value in knowing the meaning of those symbols (rather than giving them away with a related character code).

**Comment by ways2read** [comment]
I added the shortcode for  `\union` to generate the cup symbol via Word/ Options / Proofing / Maths autocorrect. This works when I type the characters but not when I build up to professional display.

**Reply by brichwin** [reply]
Hmmm... I wonder why it didn't work for you. Can you elaborate on how you created the new Math AutoCorrect entry (how did you generate the union symbol for the replace with column) and how you were creating the expression?

I created math expression using \cup, converted it, and then copied the union symbol from there when making union as a new Math AutoCorrect entry. 

After creating the new entry, I was able do press Alt+=, type `a\union b`, and then convert that into professional and the union symbol appears as desired. I have configured:
 * auto convert expressions into professional turned to disabled (unchecked), and
 * disabled all of the Math AutoCorrect codes

Here is an animated GIF of what happens when I try it (you may have to click on it to have it begin or reload the page - it only loops twice.): 
![mathAutoCorrectAliasUnionForCup](https://github.com/user-attachments/assets/b4c0727a-da72-413b-8a15-878d70b5e28b)

**Comment by ejpyattpsu** [comment]
Some other possible additions:

- For capital Greek letters (i.e. \Alpha to \Omega) I would add a variant (\alphacap to \omegacap). This distinguishes them from lowercase \alpha to \omega.
- See also \fraktura (𝔞) vs. \frakturA (𝔄) (suggesting \frakturcapa) along with \script... (script letters),  \double or \real (double strick letters) 
- Similarly \Uparrow indicates a double up arrow (vs. \uparrow for single up arrow) Should we also have something like \dbuparrow for the double up arrow.
- Looks like a lot of the negated symbols are missing shortcuts, but all the negated symbols should have their own codes 
(e.g. \notexist or \negexist or LaTeX \nexist for ∄ vs.  \exist for ∃) 
- Add \join for \bowtie
- Add \qed for QED square


FYI - I believe a lot of the original codes are coming from [LaTeX.](https://www.cmor-faculty.rice.edu/~heinken/latex/symbols.pdf) Is it worth documenting this for upper level math students? I know some screen reader users have learned raw LaTeX.

Hope this helps.
Elizabeth

**Comment by jenmarsala** [comment]
The following Wiki page contains the above codes and others in a more organized format - and also includes suggested codes for symbols which currently are lacking any Math AutoCorrect codes:
https://github.com/daisy/math-a11y/wiki/Proposed-New-Math-AutoCorrect-Codes-for-Commonly-Used-Symbols
